test/factories/host_related.rb

```diff
@@ -303,8 +303,11 @@ def set_nic_attributes(host, attributes, evaluator)
       association :parent, :factory => :hostgroup
     end
 
-    trait :with_puppetclass do
+    trait :with_environment do
       environment
+    end
+
+    trait :with_puppetclass do
```

Comment:
This can inherit from `with_environment` so as not to change its meaning.  It does depend on the environment being available, after all.